PubnativeLite/PubnativeLite/Ad Presenter/PNLiteAdPresenterDecorator.m

```diff
@@ -65,17 +65,24 @@ - (instancetype)initWithAdPresenter:(HyBidAdPresenter *)adPresenter
 #pragma mark HyBidAdPresenterDelegate
 
 - (void)adPresenter:(HyBidAdPresenter *)adPresenter didLoadWithAd:(UIView *)adView {
-    [self.adTracker trackImpression];
-    [self.adPresenterDelegate adPresenter:adPresenter didLoadWithAd:adView];
+    
+    if (self.adPresenterDelegate && [self.adPresenterDelegate respondsToSelector:@selector(adPresenter:didLoadWithAd:)]) {
```

Comment:
In which situation would the presenter not have the **didLoadWithAd** function? Same for all the ones down.

Reply:
It is mainly checking for the delegate is whether nil or not.